src/features/device-heartbeat/index.ts

```diff
@@ -256,14 +256,13 @@ export class DeviceOnlineStateManager extends EventEmitter<{
 				},
 				body,
 			});
-			return true;
 		} catch ($err) {
 			err = $err;
 			captureException(
 				$err,
-				'DeviceStateManager: Error updating the API with the device new state.',
+				`DeviceStateManager: Error updating the API state of device ${deviceId} to ${newState}.`,
 			);
-			return false;
+			throw err;
```

Comment:
If setting the DB value to Online failed then we need to rethrow at
See: https://github.com/balena-io/open-balena-api/blob/9a29bc63e70973d72738c2e0fdbf478d84705b7a/src/features/device-heartbeat/index.ts#L442
so that the Redis state is not set to Online, otherwise the device's DB record  will be stuck on its previous heartbeat state forever.